glossary.asciidoc

```diff
@@ -120,22 +132,42 @@ TODO change for clarity
 HD wallet seed::
     HD wallet seed or root seed is a potentially-short value used as a seed to generate the master private key and master chain code for an HD wallet.
 
+Homestead::
+  The second development stage of Ethereum, launched in March 2016 at block #1,150,000.
+
+Ice Age::
+  A hard fork of Ethereum at block #200,000 to introduce an exponential difficulty increase, motivating a transition to Proof-of-Stake.
+
 // In case of Ethereum, perhaps, includes a blockchain explorer too?
 IDE (Integrated Development Environment)::
 	An integrated user interface that combines a code editor, compiler, runtime, and a debugger.
 
 Immutable Deployed Code Problem::
   Once a contract's (or library's) code is deployed it becomes immutable. Being able to fix possible bugs and add new features is key for the software development cycle. This represents a challenge for smart contract development.
 
+Inter exchange Client Address Protocol (ICAP)::
+  An Ethereum Address encoding that is partly compatible with the International Bank Account Number (IBAN) encoding, offering a versatile, checksummed and interoperable encoding for Ethereum Addresses. ICAP addresses can encode Ethereum Addresses or common names registered with an Ethereum name registry.
+
 Internal transaction (also "message")::
     A transaction sent from a contract account to another contract account or an EOA.
 
+Key Derivation Function (KDF)::
+  Also known as a password stretching algorithm, it is used by keystore format which to protect against brute-force, dictionary, or rainbow table attacks against the passphrase encryption. It repeatedly hashes the passphrase.
+
+Keystone File::
```

Comment:
Sorry but, are you refering to here to Keystore file?

Reply:
ya, it's a typo